service/lib/agama/registration.rb

```diff
@@ -100,6 +104,18 @@ def on_change(&block)
 
   private
 
+    attr_reader :software
+
+    def product
+      software.product
+    end
+
+    # E.g., "ALP-Dolomite-1-x86_64"
+    def target_distro
+      v = version.to_s.split(".").first || "1"
```

Comment:
I think that we should at least log if version parameter is missing ( aka default 1 ) otherwise it can cause us some troubles later when we debug it